packages/api/tests/lambdas/test-cw-sf-event-to-db-records.js

```diff
@@ -0,0 +1,202 @@
+    t.is(fetchedExecution.collectionId, 'my-collection___5');
+    t.is(fetchedExecution.status, 'running');
+    t.is(fetchedExecution.createdAt, 122);
+    t.deepEqual(fetchedExecution.originalPayload, { key: 'my-payload' });
+  } catch (err) {
+    t.fail('Failed to fetch execution');
+  }
+});
+
+test('saveExecutionToDb() does not throw an exception if storeExecutionFromCumulusMessage() throws an exception', async (t) => {
+  const { cumulusMessage } = t.context;
+
+  // Because state_machine is missing, generating this execution record will fail
+  delete cumulusMessage.cumulus_meta.state_machine;
+
+  try {
+    await saveExecutionToDb(cumulusMessage);
+    t.pass();
+  } catch (err) {
+    t.fail(`Exception should not have been thrown, but caught: ${err}`);
+  }
+});
+
+test('saveGranulesToDb() saves a granule record to the database', async (t) => {
+  const { cumulusMessage, granuleModel } = t.context;
+
+  const stateMachineName = randomString();
+  const stateMachineArn = `arn:aws:states:us-east-1:1234:stateMachine:${stateMachineName}`;
+  cumulusMessage.cumulus_meta.state_machine = stateMachineArn;
+
+  const executionName = randomString();
+  cumulusMessage.cumulus_meta.execution_name = executionName;
+  const executionArn = `arn:aws:states:us-east-1:1234:execution:${stateMachineName}:${executionName}`;
+
+  const granuleId = randomString();
+  const files = [fakeFileFactory({ size: 250 })];
+  const granule = fakeGranuleFactoryV2({
+    files,
+    granuleId
+  });
+  delete granule.version;
+  delete granule.dataType;
+  cumulusMessage.payload.granules = [granule];
+
+  await saveGranulesToDb(cumulusMessage);
+
+  const fetchedGranule = await granuleModel.get({ granuleId });
+  const expectedGranule = {
+    ...granule,
+    collectionId: 'my-collection___5',
+    execution: `https://console.aws.amazon.com/states/home?region=us-east-1#/executions/details/${executionArn}`,
+    productVolume: 250,
+    status: 'running',
+    createdAt: 122,
+    error: {},
+    timeToArchive: 0,
+    timeToPreprocess: 0,
+    duration: fetchedGranule.duration,
+    timestamp: fetchedGranule.timestamp,
+    updatedAt: fetchedGranule.updatedAt
+  };
+  t.deepEqual(fetchedGranule, expectedGranule);
+});
+
+test.serial('saveGranulesToDb() does not throw an exception if storeGranulesFromCumulusMessage() throws an exception', async (t) => {
+  const { cumulusMessage } = t.context;
+
+  const storeGranuleStub = sinon.stub(Granule.prototype, 'storeGranulesFromCumulusMessage')
+    .callsFake(() => {
+      throw new Error('error');
+    });
+
+  try {
+    await saveGranulesToDb(cumulusMessage);
+    t.pass();
+  } catch (err) {
+    t.fail(`Exception should not have been thrown, but caught: ${err}`);
+  } finally {
+    storeGranuleStub.restore();
+  }
+});
+
+test('The cw-sf-event-to-db-records Lambda function creates execution, granule, and PDR records', async (t) => {
```

Comment:
It only creates execution and granule records, not PDRs.

Reply:
Yeah, got ahead of myself since it will also create PDRs. I can update